### PR TITLE
Bump rails & ruby versions

### PR DESCRIPTION
**Pull request by gidsg**
### Context
Required to apply security patches

### Changes proposed in this pull request
Bump rails & ruby versions 
also enforce `govuk-lint` in Travis and fix resulting errors

### Guidance to review
App should work as before.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ end
 
 ruby File.read('.ruby-version').chomp
 
-gem 'rails', '5.1.4'
+gem 'rails', '5.1.5'
```
Comment (danmitchell-) [inline Gemfile]: Lets change this to `gem 'rails', '~> 5.1.5'`

File: .travis.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ before_script:
 
 script:
   - bundle exec rake spec
-  - bundle exec govuk-lint-ruby app spec
+  - bundle exec govuk-lint-ruby app spec --format clang
```
Comment (danmitchell-) [inline .travis.yml]: Can we lint more?

 ```
- bundle exec govuk-lint-ruby app config db lib spec --format clang
 - bundle exec govuk-lint-sass app/assets/stylesheets
```

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 language: ruby
 rvm:
-- 2.4.2
+- 2.4.3
```
Comment (danmitchell-) [inline .travis.yml]: I don't think we need to specify this at all. Travis will read the `.ruby-version` https://docs.travis-ci.com/user/languages/ruby/#Using-.ruby-version

File: .travis.yml (hunk 2 above)
Comment (gidsg) [reply .travis.yml]: `govuk-lint-sass` throws up many errors. Can we add this in a separate PR?

Comment (danmitchell-) [reply .travis.yml]: 100%. I'll do that this afternoon 👍

**Pull request merged**